### ISSUE TITLE
Encountered obviously incorrect typing error [E0308]

### ISSUE THREAD

**Issue by Phosphorus15** [intellij-rust/intellij-rust#4241]
## Environment

* **IntelliJ Rust plugin version:** 0.2.102.2130-192
* **Rust toolchain version:** stable-x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2019.2
* **Operating system:** Windows 10 Pro.

## Problem description
Following codes raise an `mismatched type [E0308]` error
```rust
use std::fs::File;
use std::io::Read;
use syn::*;

fn main() {
    let f = "<file path>";
    let mut file = File::open(f).unwrap();
    let mut content = String::new();
    file.read_to_string(&mut content);

    let ast = syn::parse_file(&content).unwrap();
    syn::visit::visit_file(&mut FnVisitor::new(), &ast);
}
```
in that of `&ast` (the last line of `main` function), which claims "expected `&File`, found `&File`".
The actual type of `ast` is `syn::file::File` and the `visit_file` function expect a `syn::file::File` as well.
Codes above can be successfully compiled but `IntelliJ-Rust` seems it can't figure it out correctly.
I suspect that the plugin mistakes `ast` to have type `std::fs::File`

## Steps to reproduce
`Cargo.toml` depedencies
```ini
[dependencies]
quote = "0.6"

[dependencies.syn]
version = "0.15"
default-features = false
features = ["full", "printing", "parsing", "extra-traits", "visit", "clone-impls"]
```

link to [Cargo Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8e5e64fb512e44973a6051de3f887424)

**Comment by Undin** [comment]
I can't reproduce this error now.
So let's close the issue.
@Phosphorus15 if it's reproducible for you yet, don't hesitate to ping us
